build-tools-integration-tests/src/test/java/org/projectnessie/buildtools/TestPublishedPoms.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.projectnessie.buildtools;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+import java.util.List;
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.resolution.DependencyResult;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/** Resolves all published Nessie jar artifacts and lets Maven fail on any issue in a pom. */
+public class TestPublishedPoms {
+
+  static RepositorySystem repositorySystem;
+  static RepositorySystemSession repositorySystemSession;
+  static List<RemoteRepository> repositories;
+  static String nessieVersion;
+
+  @BeforeAll
+  static void setup() {
+    nessieVersion = System.getProperty("nessie.version");
+    assertThat(nessieVersion).as("System property nessie.version").isNotNull();
+    repositorySystem = buildRepositorySystem();
+    repositorySystemSession = newSession(repositorySystem);
+    repositories =
+      singletonList(
+        new RemoteRepository.Builder(
+          "maven-central", "default", "https://repo1.maven.org/maven2/")
+          .build());
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+    "nessie-client",
```

Comment:
Cannot read the properties files in ../gradle/, because those contain _all_ Nessie projects, not just only the ones publishing jars.